### DISCUSSION TITLE
How do you represent the fact that a cemetery changed its name?

### DISCUSSION THREAD

**Discussion by Stephen-Gates** [muninn/graves#41] [Q&A]
The `Brisbane General Cemetery` changed its name to `Toowong Cemetery`. I'm still trying to find the date of the name change.

How is this best represented? 

Should there be, either:

1.  one cemetery record with an "also known as" property and value  ("also known as" is used on the [Toowong Cemetery Wikidata entry](https://www.wikidata.org/wiki/Q7824339))

2. two cemetery records, `Brisbane General Cemetery` from 1871 to ????, and `Toowong Cemetery` from ???? to null endDate

If the answer is 1., then is an "also known as" property needed on [Dead-people-place](https://rdf.muninn-project.org/ontologies/graves-en.html#term_Dead_people_place)?

**Comment by rwarren2** [comment]
_The thing and the name of the thing are different things._

Most of the examples in the ontology use siteName (#23) or rdfs:label, both of which keep people happy with simplicity but don't deal well with changes to appellation. The nearest property to "also known as" is skos:altLabel which doesn't give you temporal data but is easy to search; this is also what Wikidata is using:
![Untitled](https://user-images.githubusercontent.com/11589124/147854194-6ab90e83-3117-4aeb-8327-7544fcfb0607.png)

In this case are you trying to deal with its change in legal organization, something like the [W3 Organization Ontology](https://www.w3.org/TR/vocab-org/) or [FIBO](https://github.com/edmcouncil/fibo) are the answer but that might be overkill.

Graves imports the [TIME Ontology](https://www.w3.org/TR/owl-time/) and aligns it with the [PROV Ontology](https://www.w3.org/TR/prov-o/) as per the [recommended statements](https://www.w3.org/TR/owl-time/#time-prov) for temporal data. [skos-xl](https://www.w3.org/TR/skos-reference/skos-xl.html) provides a basis for a label-as-an-instance but restricts the [skos-xl:literalForm](http://www.w3.org/2008/05/skos-xl#literalForm) property to a singleton which annoyingly prevents multilingual applications. Thus my suggestion is that I create a new class called graves:FeatureName that is a [prov:Activity](https://www.w3.org/TR/prov-o/#Activity) (subclassed from [time:TemporalEntity](https://www.w3.org/TR/owl-time/#time:TemporalEntity)) and a [skos-xl:Label](https://www.w3.org/TR/skos-reference/skos-xl.html#Label). This example is in verbose xml/rdf but you get the idea:

```
<graves:Cemetery>
 <rdfs:label xml:lang="en">Toowong Cemetery</rdfs:label>
 <rdfs:label xml:lang="en">Brisbane General Cemetery</rdfs:label>
 <skos:prefLabel>
  <graves:FeatureName rdf:about="#brisbane">
   <rdfs:label xml:lang="en">Brisbane General Cemetery</rdfs:label>
   <skos-xl#literalForm xml:lang="en">Brisbane General Cemetery</skos-xl#literalForm>
   <prov:startedAtTime rdf:datatype="http://www.w3.org/2001/XMLSchema#gYear">1871</prov:startedAtTime>
   <time:before rdf:resource="#toowong"/>
  </graves:FeatureName>
 </skos:prefLabel>
 <skos:prefLabel>
  <graves:FeatureNamer df:about="#toowong">
   <time:after rdf:resource="#brisbane"/>
   <rdfs:label xml:lang="en">Toowong Cemetery</rdfs:label>
   <skos-xl#literalForm xml:lang="en">Toowong Cemetery</skos-xl#literalForm>
  </graves:FeatureName>
 </skos:prefLabel>
...
</graves:Cemetery>
```

Two items of import here:
1. The proposed graves:FeatureName is just semantic glue for  [prov:Activity](https://www.w3.org/TR/prov-o/#Activity) (and [time:TemporalEntity](https://www.w3.org/TR/owl-time/#time:TemporalEntity) by the graves ontology) and a [skos-xl:Label](https://www.w3.org/TR/skos-reference/skos-xl.html#Label). You can get the same result with your own construct but it is redundant.
2. The concurrent use of [prov:Activity](https://www.w3.org/TR/prov-o/#Activity) and [time:TemporalEntity](https://www.w3.org/TR/owl-time/#time:TemporalEntity) gives you a very large number of options for temporal data: You get [startedAtTime](https://www.w3.org/TR/prov-o/#startedAtTime) and [endedAtTime](https://www.w3.org/TR/prov-o/#endedAtTime) for the label using [W3 XML Schemas](https://www.w3.org/TR/xmlschema-2/) with their support for truncated date times. You also get all of the power of the [TIME Ontology](https://www.w3.org/TR/owl-time/) for building complex and relative time descriptions, which include [time:before](https://www.w3.org/TR/owl-time/#time:before) and [time:after](https://www.w3.org/TR/owl-time/#time:after); this allows you to commit to a sequence of nomenclatures without having to nail down specific dates.

SPARQL and XSD have very specific ideas about how to evaluate truncated representation and not all Inferenceable time comparisons will be resolvable using the partial data that you have, but it fixes most of your day to day problems such as ordering and name display.

Any thoughts?

PS: There can only be one [skos-xl#literalForm](https://www.w3.org/TR/skos-reference/skos-xl.html#literalForm) per [skos-xl:Label](https://www.w3.org/TR/skos-reference/skos-xl.html#Label) because of an owl:Restriction which causes problems for multilingual applications. This is at odds with S14: "[A resource has no more than one value of skos:prefLabel per language tag](https://www.w3.org/TR/skos-reference/#L1567)". In this case, I'm simply using multiple rdfs:label properties.

**Comment by rwarren2** [comment]
Current thinking:
```

  <owl:Class rdf:ID="TemporalName">
   <rdfs:subClassOf rdf:resource="http://www.w3.org/2006/time#"/>
   <rdfs:subClassOf rdf:resource="http://www.w3.org/2008/05/skos-xl"/>
   <rdfs:label xml:lang="en">Name</rdfs:label>
   <rdfs:label xml:lang="fr">Nom</rdfs:label>
   <rdfs:label xml:lang="de">Name</rdfs:label>
   <skos:definition xml:lang="en">The name of a feature, place or thing that has a temporal validity.</skos:definition>
   <skos:definition xml:lang="fr">Le nom d'un toponyme, d'un lieu ou d'une chose qui a une validité temporelle.</skos:definition>
   <skos:definition xml:lang="de">Der Name eines Merkmals, Ortes oder Dings, das eine zeitliche Gültigkeit hat.</skos:definition>
   <rdfs:isDefinedBy rdf:resource="https://rdf.muninn-project.org/ontologies/graves-ONTOLOGY_DATE"/>
   <rdf:type rdf:resource="http://www.w3.org/2004/02/skos/core#Concept"/>
   <void:inDataset rdf:resource="#datasetdefinition"/>
  </owl:Class>
```

**Comment by Stephen-Gates** [comment]
Thanks for your very detailed reply. As someone new to linked data, I do not feel qualified to comment on what might be a good solution but what you describe using Time and Prov seems very powerful.  I suspect my contributions will be limited to raising issues, questions, and test cases. I'm sorry I can't be more helpful on implementations. I'll continue my learning so perhaps I may be able to contribute in a more meaningful way. 

Due to my lack of research, I'm not sure if this example is an alias, simple name change, or something more significant in the operations of the Cemetery (e.g. change in Administrating body - which did happen but I'm not sure if the name change is linked to that event). I'll continue my research to try clarify. At the moment, I'm leaning towards an alias, until I find some evidence.

**Comment by rwarren2** [comment]
OK. [skos:altLabel](http://www.w3.org/2004/02/skos/core#altLabel) is the way to go for you then. 

I'm going to create the TemporalName in v1.5 to fix a few other corner cases.
